keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+support some Kubernetes functionality like e.g. Services, I am willing to give
+that up for performance.
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster operator I wish to isolate workloads based on
+namespaces and network access by connecting to only a non- Cluster Default PodNetwork.
+Those workloads should have the same level of Kubernetes functionality:
+Services, NetworkPolicies, access to Kubernetes API.
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As Cluster operator I wish to manage my cluster’s bandwidth usage on a per-pod
+basis, simultaneously preserving all other Pod networking configuration the same
+across those Pods.
+
+<p align="center">
+  <img src="mn-story-10.png?raw=true" alt="multi-network story 10 per-pod connection differentiation"/>
+</p>
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will be in its own design
+section of this KEP that will be defined and implemented incrementally.
+
+#### Phase I (base API and reference in Pod)
+1. Multi-network should not change the behavior of existing clusters where no 
+multi-network constructs are configured/used.
+2. Multi-network needs to be represented explicitly as an API construct in
+Kubernetes as it fundamentally ties in with other APIs in the Kubernetes
+networking model, extending their functionality.
+   * We will introduce an PodNetwork API object that represents the existing
+     infrastructure’s networking.
```

Comment:
How do you solve the boostrap problem, Nodes are created and then Pods, when is the PodNetwork object created and by whom?
what is the lifecycle of the PodNetwork object?
how does it interact with the lifecycle of the Nodes and Pods objects?
How is the infrastructure represented?
How does it handle network partitions in the infrastructure? What happens if NodeA and talk with Node B but not viceversa?

Reply:
Most of these are answered below. Regarding
>How does it handle network partitions in the infrastructure? What happens if NodeA and talk with Node B but not viceversa?

 This KEP will not define/explain any infrastructure relations. The PodNetwork is just an abstraction for that, and it is up to the implementation on how strongly it is coupled with the infra.